routes/actions.js

```diff
@@ -347,4 +348,10 @@ function sendRegistrationForm(req, res, formInfo) {
   });
 }
 
+function returnToPath(reqOrig, res) {
+    let reqUrl = reqOrig.originalUrl;
+    let query = url.parse(reqUrl, true).query.returnTo;
+    res.redirect(query);
```

Comment:
This doesn't check against external vs. internal URL, so it could be used to redirect the user to another website. For security reasons, we have to prevent that, e.g., by ensuring that the returnTo value is a path.